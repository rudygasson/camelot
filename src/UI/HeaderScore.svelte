<script>
  export let score = 0;
</script>

<div role="cell" class="score">
  <slot></slot>
  <div class="score__value">{score}</div>
</div>

<style>
  .score {
    height: 2rem;
    display: flex;
    justify-content: space-around;
    align-items: center;
    width: 4.4rem;
  }

  .score__value {
    margin: 0;
    font-size: 1.5rem;
  }
</style>
