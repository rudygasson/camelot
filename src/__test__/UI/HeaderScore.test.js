/**
 * @jest-environment jsdom
 */

import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/svelte";

import HeaderScore from "../../UI/HeaderScore.svelte";

describe("Header Score component", () => {
  it("shows passed value prop as score", () => {
    const { container } = render(HeaderScore, { score: 34 });
    expect(container.querySelector(".score__value").textContent).toEqual("34");
  });
  
  it("shows 0 as default value", () => {
    const { container } = render(HeaderScore);
    expect(container.querySelector(".score__value").textContent).toEqual("0");
  });

  it("does not change markup", () => {
    const { container } = render(HeaderScore, { score: 42 });
    expect(container).toMatchSnapshot();
  });
});
