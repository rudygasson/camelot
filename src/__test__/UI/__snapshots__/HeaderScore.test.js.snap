// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Score component does not change markup 1`] = `
<body>
  <div>
    <div
      class="score svelte-1s4e1st"
      role="cell"
    >
       
      <div
        class="score__value svelte-1s4e1st"
      >
        42
      </div>
    </div>
  </div>
</body>
`;
